packages/atomic/src/utils/accessibility-utils.ts

```diff
@@ -49,12 +52,83 @@ export class AriaLiveRegionController implements ReactiveController {
   }
 }
 
-//TODO: Reimplement to fit Lit
+// TODO: see if this can be reimplemented as a mixin
 export class FocusTargetController {
-  public setTarget(_el: HTMLElement) {}
-  public focusAfterSearch() {}
-  public focusOnNextTarget() {}
-  public async focus() {}
+  private bindings: AnyBindings;
+  private lastSearchId?: string;
+  private element?: HTMLElement;
+  private onFocusCallback?: Function;
+  private doFocusAfterSearch = false;
+  private doFocusOnNextTarget = false;
+
+  constructor(
+    private component: InitializableComponent<AnyBindings> & LitElement
+  ) {
+    this.bindings = component.bindings;
+    this.handleComponentRenderLoop();
+  }
+
+  public setTarget(el: Element | undefined) {
+    if (!el) {
+      return;
+    }
+    this.element = el as HTMLElement;
+    if (this.doFocusOnNextTarget) {
+      this.doFocusOnNextTarget = false;
+      this.focus();
+    }
+  }
+
+  public async focus() {
+    await defer();
+    this.element?.focus();
+    this.onFocusCallback?.();
+  }
+
+  public focusAfterSearch() {
+    this.lastSearchId = this.bindings.store.getUniqueIDFromEngine(
+      this.bindings.engine
+    );
+    this.doFocusAfterSearch = true;
+    return new Promise((resolve) => (this.onFocusCallback = resolve));
+  }
+
+  public focusOnNextTarget() {
+    this.doFocusOnNextTarget = true;
+    return new Promise((resolve) => (this.onFocusCallback = resolve));
+  }
+
+  public disableForCurrentSearch() {
+    if (
+      this.bindings.store.getUniqueIDFromEngine(this.bindings.engine) !==
+      this.lastSearchId
+    ) {
+      this.doFocusAfterSearch = false;
+    }
+  }
+
+  private handleComponentRenderLoop() {
+    // @ts-expect-error - Acesssing protected method
+    const originalUpdated = this.component.updated;
+    // @ts-expect-error - Overridding protected method
+    this.component.updated = async (_changedProperties) => {
```

Comment:
This is a good place where a reactive component would be required to prevent having to override protected method
```ts
export class FocusTargetController implements ReactiveController {
  // ...
  constructor(
    private host: ReactiveControllerHost,
    bindings: AnyBindings
  ) {
    this.host = host;
    this.bindings = bindings;
    this.host.addController(this);
  }

  hostUpdated() {
    // ...
  }
}
```

Reply:
I don't think it has to be converted into a mixin. 
Using a reactive controller will do the work